From: Prarit Bhargava <prarit@redhat.com>
Date: Thu, 20 Nov 2008 10:41:21 -0500
Subject: [acpi] add systems to GPE register blacklist
Message-id: 20081120154121.32739.12528.sendpatchset@prarit.bos.redhat.com
O-Subject: [RHEL5.3 PATCH] Add all HP xw systems to GPE register blacklist
Bugzilla: 471341
RH-Acked-by: Peter Martuccelli <peterm@redhat.com>
RH-Acked-by: Pete Zaitcev <zaitcev@redhat.com>
RH-Acked-by: Brian Maly <bmaly@redhat.com>

This is a follow-up to 456638.  In that BZ it was noted that the legacy value
for the GPE register base differed from that of the 64-bit value.  In these
cases, the 64-bit value is to be used for the GPE registers.

However, on the HP xw series systems the 64-bit value is incorrect and leads
to errors being displayed on the console (and in some cases endless errors on
the console).

I had blacklisted 4 systems -- it turns out that almost all of the HP xw
systems hit this bug.

Add all HP xw series systems to the blacklist.

Sucessfully compiled and tested by me.

Resolves BZ 471341.

diff --git a/drivers/acpi/events/evgpeblk.c b/drivers/acpi/events/evgpeblk.c
index a5093f3..1b9852f 100644
--- a/drivers/acpi/events/evgpeblk.c
+++ b/drivers/acpi/events/evgpeblk.c
@@ -1076,34 +1076,10 @@ static int hp_gpe_use_32_bit(struct dmi_system_id *d)
 static struct dmi_system_id acpi_hp_gpe_table[] = {
 	{
 	 .callback = hp_gpe_use_32_bit,
-	 .ident = "HP xw 4600",
+	 .ident = "HP xw",
 	 .matches = {
 		     DMI_MATCH(DMI_BOARD_VENDOR, "Hewlett-Packard"),
-		     DMI_MATCH(DMI_PRODUCT_NAME, "HP xw4600 Workstation"),
-	 	    },
-	},
-	{
-	 .callback = hp_gpe_use_32_bit,
-	 .ident = "HP xw 4800",
-	 .matches = {
-		     DMI_MATCH(DMI_BOARD_VENDOR, "Hewlett-Packard"),
-		     DMI_MATCH(DMI_PRODUCT_NAME, "HP xw4800 Workstation"),
-	 	    },
-	},
-	{
-	 .callback = hp_gpe_use_32_bit,
-	 .ident = "HP xw 8600",
-	 .matches = {
-		     DMI_MATCH(DMI_BOARD_VENDOR, "Hewlett-Packard"),
-		     DMI_MATCH(DMI_PRODUCT_NAME, "HP xw8600 Workstation"),
-	 	    },
-	},
-	{
-	 .callback = hp_gpe_use_32_bit,
-	 .ident = "HP xw 9400",
-	 .matches = {
-		     DMI_MATCH(DMI_BOARD_VENDOR, "Hewlett-Packard"),
-		     DMI_MATCH(DMI_PRODUCT_NAME, "HP xw9400 Workstation"),
+		     DMI_MATCH(DMI_PRODUCT_NAME, "HP xw"),
 	 	    },
 	},
 	 {}
